### ISSUE TITLE
Using Controller subdirectory for Routing Resource

### ISSUE THREAD

**Issue by leek** [FriendsOfSymfony/FOSRestBundle#189]
I am attempting to load all controllers under a subdirectory as `type: rest` in my Route configuration similar to how it is done in the `SensioFrameworkExtraBundle` [here](http://symfony.com/doc/2.0/bundles/SensioFrameworkExtraBundle/annotations/routing.html#activation):

```yml
# routing.yml

foobar_api:
    resource: "@PWFoobarBundle/Controller/Api/"
    type:     rest
```

Is anything like this possible? Is there another way of setting up routes from within a subdirectory of `Controller/`? I've tried any syntax I could think of, even down to specifying individual controller classes within the subdirectory and none seem to work.

**Comment by Soullivaneuh** [comment]
Great :+1: for this.

Having this kind of configuration:

```yaml
api:
    type:     rest
    resource: "@AppBundle/Controller/"
    prefix:   /api
```

Would be very painless than defined a resource per controller, especially for huge API.

**Comment by Ener-Getick** [comment]
Fixed by https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1339

**Comment by magnetik** [comment]
I just tested and it seems that it's not working.

I'm trying to load controllers with the following code:

```
api:
    resource: "@AppBundle/ApiController/"
    type:     rest

```
Firstly `FOS\RestBundle\Routing\Loader\RestRouteLoader` is created/tagged before the `DirectoryRouteLoader` in `routing.xml`, and the `supports` methods does not exclude the case when the resource is a directory.

Secondly it seems that calling `is_dir($resource)` will fail, because `"@AppBundle/ApiController"` is not a directory: it needs some resolution to a path first. 

Not sure if I should create a new bug instead of reopening this one. I can work on a PR.

**Comment by xabbuh** [comment]
@magnetik Can you try the latest version of the `1.8` or `2.0` branch for this please? We did not tag another beta yet after fixing this bug.

**Comment by magnetik** [comment]
Ho, indeed it is working now. I was using `~2.0@dev` and I though it was giving me head of the 2.0 branch.

Thanks

**Comment by xabbuh** [comment]
@magnetik You will then end up with `2.1.x-dev` (admittedly we didn't merge up `2.0` to `master` for some time). You can use the `~2.0.0@dev` constraint instead.

**Comment by Ener-Getick** [comment]
@magnetik @xabbuh I merged 2.0 into master.
